### PR TITLE
added "title" columns

### LINKED ISSUE #7361: Entities should have an optional display name
## Feature Suggestion

I would like service metadata to include an optional display name which could be used to record a more human friendly name. 

The limitations around use of whitespace, and the compactness of the table in the main service catalog, can render service names difficult to read. They're also not very human friendly due to the use of hyphens.

Screenshot 2021-09-28 at 13 24 46

## Possible Implementation

The human friendly name would be used in the catalog list and in the header of the main entity page. It would not need to be unique.

### PR DESCRIPTION
**Pull request by alexrybch**
Signed-off-by: Alex Rybchenko <arybchenko@box.com>

## Hey, I just made a Pull Request!

Added functions to render recently added "title" field from entity metadata in EntityTable and CatalogTable components.
image

Closes #7361 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Makes sense, thanks! 👍 

Just need to add some changesets!

**Comment by alexrybch** [comment]
added changeset :)

**Comment by freben** [comment]
@Rugvip An alternative would be to have one column, that shows either

- the name, if that's all we have, like before
- the title if it exists, with the name as a tooltip or so (so you can kind of get at both)

There's a larger picture to this of course. For example in a page header, would we do the exact same? Or perhaps the title as the main title and the name as the subtitle? It's of course important to be able to conveniently be able to see both somehow

**Comment by alexrybch** [comment]
@freben I like your idea with one column, will do the update!

**Comment by dhenneke** [comment]
> @Rugvip An alternative would be to have one column, that shows either
> 
> * the name, if that's all we have, like before
> * the title if it exists, with the name as a tooltip or so (so you can kind of get at both)

It would be nice if the `EntityRefLink` component could magically display the title if available. This way it would be consistent everywhere. But sadly, the `EntityRefLink` only has the entity ref and would require a rest call to get the title information...

**Comment by alexrybch** [comment]
I removed new column creators and updated `EntityRefLink` to accept `title' prop to override default entity title (default name is shown on hover). Also updated existing `createNameColumn` functions to pass `metadata.title' as title prop to `EntityRefLink`.
image

**Review by Rugvip — COMMENTED** [review]
Looks good again, and still just needs a changeset x)

For fixing the tests you can use `renderInTestApp` from `@backstage/test-utils` and pass in this as an option:

```ts
mountedRoutes: {
  '/catalog/:namespace/:kind/:name/*': entityRouteRef,
},
```

**Comment by alexrybch** [comment]
> Looks good again, and still just needs a changeset x)
> 
> For fixing the tests you can use `renderInTestApp` from `@backstage/test-utils` and pass in this as an option:
> 
> ```ts
> mountedRoutes: {
>   '/catalog/:namespace/:kind/:name/*': entityRouteRef,
> },
> ```

my bad, added changeset :)
thanks for help with tests!

File: .changeset/few-sloths-raise.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog': minor
+'@backstage/plugin-catalog-react': minor
```
Comment (Rugvip) [inline .changeset/few-sloths-raise.md:3]: Since this is a non-breaking change and we're on `0.x` we can make these `patch`

**Comment by dtuite** [comment]
I released the builds on this PR just now.

**Comment by alexrybch** [comment]
rebased and updated tests, please review

**Review by benjdlambert — APPROVED** [review]

**Review by emmaindal — APPROVED** [review]

**Pull request merged**